models/salesforce/intermediate/int_salesforce__date_spine.sql

```diff
@@ -25,7 +25,12 @@ with spine as (
 
     {% if execute %}
     {% set last_date_query %}
-        select  max( created_date ) as max_date from {{ source('salesforce', 'opportunity') }}
+        {% if var('salesforce__lead_enabled', True) %}
+            select  max( created_date ) as max_date from {{ source('salesforce', 'lead') }}
+        {% else %}
+        select  coalesce(max( created_date ), '2024-01-01') as max_date from {{ source('salesforce', 'opportunity') }}
```

Comment:
ah because of this [line](https://github.com/fivetran/dbt_salesforce/blob/bug/use-lead-for-spine-end-date/models/salesforce/intermediate/int_salesforce__date_spine.sql#L40), we end up choosing the current date anyways!

but perhaps this whole snippet is ultimately kinda roundabout and unnecessary -- is the last_date ever going to be greater than the current date, or should we just automatically choose the current_date? changing it entirely might be venturing out of scope i suppose though

Reply:
FR made! https://github.com/fivetran/dbt_salesforce/issues/49